File: arrays_and_hashing/347_top_k_frequent_elements.py
# https://leetcode.com/problems/top-k-frequent-elements/

class Solution:
    def topKFrequent(self, nums: list[int], k: int) -> list[int]:
        count = {}
        for n in nums:
            count[n] = count.get(n, 0) + 1

        freq = [[] for _ in range(len(nums) + 1)]
        for n, c in count.items():
            freq[c].append(n)

        ans = []
        for i in range(len(freq) - 1, 0, -1):
            for n in freq[i]:
                ans.append(n)
                if len(ans) == k:
                    return ans

File: go/arrays_and_hashing/347_top_k_frequent_elements.go
package arraysandhashing

func TopKFrequent(nums []int, k int) []int {
	count := make(map[int]int, len(nums))
	for _, n := range nums {
		count[n]++
	}

	freq := make([][]int, len(nums)+1)
	for n, c := range count {
		freq[c] = append(freq[c], n)
	}

	res := make([]int, 0, k)
	for i := len(freq) - 1; i > 0; i-- {
		res = append(res, freq[i]...)
		if len(res) == k {
			return res
		}
	}
	return res
}
